GeoLib/PointVec.cpp

```diff
@@ -33,23 +33,55 @@ PointVec::PointVec (const std::string& name, std::vector<Point*>* points,
 	TemplateVec<Point> (name, points, name_id_map),
 	_type(type),
 	_aabb(points->begin(), points->end()),
-	_rel_eps(rel_eps)
+	_rel_eps(rel_eps*sqrt(MathLib::sqrDist(_aabb.getMinPoint(), _aabb.getMaxPoint()))),
+	_oct_tree(OctTree<GeoLib::Point, 16>::createOctTree(
+		_aabb.getMinPoint(), _aabb.getMaxPoint(), _rel_eps))
 {
 	assert (_data_vec);
-	std::size_t const number_of_all_input_pnts (_data_vec->size());
+	std::size_t const number_of_all_input_pnts(_data_vec->size());
 
-	_rel_eps *= sqrt(MathLib::sqrDist (_aabb.getMinPoint(),_aabb.getMaxPoint()));
-	makePntsUnique (_data_vec, _pnt_id_map, _rel_eps);
+	// correct the ids if necessary
+	for (std::size_t k(0); k<_data_vec->size(); ++k) {
+		if ((*_data_vec)[k]->getID() == std::numeric_limits<std::size_t>::max()) {
+			(*_data_vec)[k]->setID(k);
+		}
+	}
+
+	// add all points in the oct tree in order to make them unique
+	_pnt_id_map.resize(number_of_all_input_pnts);
+	std::iota(_pnt_id_map.begin(), _pnt_id_map.end(), 0);
+	GeoLib::Point * ret_pnt(nullptr);
+	std::size_t cnt(0);
+	for (std::size_t k(0); k<_data_vec->size(); ++k) {
+		GeoLib::Point *const pnt((*_data_vec)[k]);
+		if (! _oct_tree->addPoint(pnt, ret_pnt)) {
+			DBUG("Point %d (%f,%f,%f) *not* inserted since it is equal to "
+				"point %d (%f,%f,%f).", pnt->getID(), (*pnt)[0], (*pnt)[1], (*pnt)[2],
+				ret_pnt->getID(), (*ret_pnt)[0], (*ret_pnt)[1], (*ret_pnt)[2]);
+			DBUG("Removing point %d", pnt->getID());
```

Comment:
The two DBUG messages are very noisy when running tests. Please remove or comment.
Same in line 149ff.